src/StackExchange.Utils.Http/HttpSettings.cs

```diff
@@ -55,6 +56,16 @@ public class HttpSettings
         /// </summary>
         public TimeSpan DefaultTimeout { get; set; } = TimeSpan.FromSeconds(30);
 
+        /// <summary>
+        /// The default Proxy to use when making requests.
+        /// 
+        /// This should create a new instance of a proxy when called,
+        /// so that modifications don't affect the default (e.g.,
+        /// changing Proxy.Credentials on a builder.Proxy should
+        /// not affect the Proxy.Credentials used by default)
```

Comment:
I'm not sure I follow this comment - we aren't modifying the proxy are we?

Reply:
If you create a new Builder, and then do e.g., `builder.Proxy.Credentials = new NetworkCredentials(...)`, you are modifying the instance that came from the factory. if the factory always hands out the same instance (`() => myProxy;`), then you are modifying the default `HttpSettings` proxy through a builder instance.

Reply:
Alternatively, the DefaultProxyFactory could be removed entirely, and it could be required to set the Proxy on each builder - that works for my use case, but maybe not as neat as being able to set the default.